test/func_test/test_ctrt.py

```diff
@@ -577,3 +577,286 @@ async def new_atomic_swap_ctrt(
+        resp = await nc.deposit(acnt0, ac.ctrt_id, 10)
+        await cft.wait_for_block()
+        tx_info = await api.tx.get_info(resp["id"])
+        assert tx_info["status"] == "Success"
+
+        tok_bal = await get_tok_bal(api, acnt0.addr.b58_str, tok_id)
+        assert tok_bal == 40
+
+        deposited_tok_bal = await ac.get_tok_bal(acnt0.addr.b58_str)
+        assert deposited_tok_bal == 10
+
+    async def test_withdraw(
+        self, new_ctrt_with_tok: pv.TokenCtrtWithoutSplit, acnt0: pv.Account
+    ):
+        """
+
+        Args:
+            new_ctrt_with_tok (pv.NFTCtrt): [description]
+            acnt0 (pv.Account): [description]
+        """
+        nc = new_ctrt_with_tok
+        api = nc.chain.api
+
+        tok_id = await get_tok_id(api, nc.ctrt_id, 0)
+        ac = await pv.AtomicSwapCtrt.register(acnt0, tok_id)
+        await cft.wait_for_block()
+        assert (await ac.maker) == acnt0.addr.b58_str
+        assert (await ac.token_id) == tok_id
+
+        await nc.deposit(acnt0, ac.ctrt_id, 10)
+        await cft.wait_for_block()
+
+        tok_bal = await get_tok_bal(api, acnt0.addr.b58_str, tok_id)
+        assert tok_bal == 40
+
+        deposited_tok_bal = await ac.get_tok_bal(acnt0.addr.b58_str)
+        assert deposited_tok_bal == 10
+
+        await nc.withdraw(acnt0, ac.ctrt_id, 10)
+        await cft.wait_for_block()
+
+        tok_bal = await get_tok_bal(api, acnt0.addr.b58_str, tok_id)
+        assert tok_bal == 50
+
+        deposited_tok_bal = await ac.get_tok_bal(acnt0.addr.b58_str)
+        assert deposited_tok_bal == 0
+
+    async def test_supersede(
+        self, new_ctrt: pv.TokenCtrtWithoutSplit, acnt0: pv.Account, acnt1: pv.Account
+    ):
+        """
+        test_supersede tests the method supersede.
+
+        Args:
+            new_ctrt (pv.NFTCtrt): The fixture that registers a new token contract.
+            acnt0 (pv.Account): The account of nonce 0.
+            acnt1 (pv.Account): The account of nonce 1.
+        """
+        nc = new_ctrt
+        api = nc.chain.api
+
+        assert (await nc.issuer) == acnt0.addr.b58_str
+
+        resp = await nc.supersede(acnt0, acnt1.addr.b58_str)
+        await cft.wait_for_block()
+        await cft.assert_tx_success(api, resp["id"])
+
+        assert (await nc.issuer) == acnt1.addr.b58_str
+
+    async def test_destroy(
+        self, new_ctrt_with_tok: pv.TokenCtrtWithoutSplit, acnt0: pv.Account
+    ):
+        """
+
+        Args:
+            new_ctrt_with_tok (pv.TokenCtrtWithoutSplit): [description]
+        """
+        nc = new_ctrt_with_tok
+        api = nc.chain.api
+
+        tok_id = await get_tok_id(api, nc.ctrt_id, 0)
+
+        tok_bal = await get_tok_bal(api, acnt0.addr.b58_str, tok_id)
+        assert tok_bal == 50
+
+        resp = await nc.destroy(acnt0, 10)
+        await cft.wait_for_block()
+        await cft.assert_tx_success(api, resp["id"])
+
+        tok_bal_acnt0 = await get_tok_bal(api, acnt0.addr.b58_str, tok_id)
```

Comment:
Can also check the token info by `/contract/tokenInfo/{tokenId}` and assert the field `total` decreases as well.

Below is a sample response
```json
{
  "tokenId": "TWuEQu6N2fMDrBdZFWY9CyhfWVECTrgmAgEAmF2yL",
  "contractId": "CF8e4uK2exZa7UTpPSLmJBXthH17TiRo4tY",
  "max": 100,
  "total": 90,
  "unity": 1,
  "description": ""
}
```